test/SILOptimizer/static_arrays.swift

```diff
@@ -117,6 +117,15 @@ func overwriteLiteral(_ x: Int) -> [Int] {
   return a
 }
 
+struct Empty { }
+
+// CHECK-LABEL: sil {{.*}}arrayWithEmptyElements{{.*}} : $@convention(thin) () -> @owned Array<Empty> {
+func arrayWithEmptyElements() -> [Empty] {
+  // CHECK:    alloc_ref
+  // CHECK:    return
+  return [Empty()]
+}
```

Comment:
Is this something we can fix in our modelling of global objects, so that empty values as part of the tail allocation take space according to their stride instead of their size? Is this also a problem for types with tail padding such that size != stride?

Reply:
Yeah, that's a good point. It would be cleaner to fix this in IRGen. I'll do it in a follow-up commit.

>  Is this also a problem for types with tail padding such that size != stride?

I don't think it's a problem because the alignment is taken into account. And the stride is just derived from size and alignment (except for the 0-size case)